integrations/operator/controllers/resources/teleport_reconciler.go

```diff
@@ -199,6 +199,22 @@ func (r TeleportResourceReconciler[T, K]) Upsert(ctx context.Context, obj kclien
 
 // Delete is the TeleportResourceReconciler of the ResourceBaseReconciler DeleteExertal
 func (r TeleportResourceReconciler[T, K]) Delete(ctx context.Context, obj kclient.Object) error {
+	// This call catches non-existing resources or subkind mismatch (e.g. openssh nodes)
+	// We can then check that we own the resource before deleting it.
+	resource, err := r.resourceClient.Get(ctx, obj.GetName())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, isOwned := checkOwnership(resource)
+	if !isOwned {
+		// The resource doesn't belong to us, we bail out but unblock the CR deletion
+		// TODO: discuss with marco and tiago if this is OK or if we prefer to block CR deletion
+		return nil
+	}
```

Comment:
I feel it's fine because the operator never managed that resource.
What we should never allow, is creating the crd if the resource already exists in teleport with a different owner.
Should we pursue that using validation hooks in the feature? Would solve a lot of problems where users assume we manage the resource but we never do

Reply:
Created https://github.com/gravitational/teleport/issues/37705

Reply:
```suggestion
	if !isOwned {
		// The resource doesn't belong to us, we bail out but unblock the CR deletion
		return nil
	}
```